website/client-old/js/controllers/inventoryCtrl.js

```diff
@@ -253,20 +253,40 @@ habitrpg.controller("InventoryCtrl",
             size:'sm'
           });
         }
-
-        // Checks if mountmaster has been reached for the first time
-        if(!user.achievements.mountMaster
-            && $scope.mountCount >= 90) {
-          User.user.achievements.mountMaster = true;
-          Achievement.displayAchievement('mountMaster');
-        }
+        handleMountAchievements();
 
       // Selecting Pet
       } else {
         User.equip({params:{type: 'pet', key: pet.key}});
       }
     }
+    function handleMountAchievements() {
+        //Mount collector
+        var retrospectiveCount = (user.achievements.mountMaster || 0) * 90 + $rootScope.countExists(user.items.mounts);
+        user.achievements.mountCollector = Math.max(retrospectiveCount, user.achievements.mountCollector || 0);
```

Comment:
Oh, now I get it. I was sure that the user that I had there was the User.user collector used (Actually I think I was right there- but then I moved this piece of code to this file and forgot to change it. Not sure about it, but it doesn't matter).